src/NavBar.ts

```diff
@@ -11,21 +11,26 @@ import { getApi } from '@/api/Api'
 export default class NavBar {
     private readonly queryStore: QueryStore
     private readonly mapStore: MapOptionsStore
-    private isIgnoreQueryStoreUpdates = false
+    private ignoreStateUpdates = false
 
     constructor(queryStore: QueryStore, mapStore: MapOptionsStore) {
         this.queryStore = queryStore
-        this.queryStore.register(() => this.onQueryStateChanged())
         this.mapStore = mapStore
-        this.mapStore.register(() => this.onQueryStateChanged())
-        window.addEventListener('popstate', () => this.parseUrlAndReplaceQuery())
+        window.addEventListener('popstate', async () => await this.updateStateFromUrl())
+    }
+
+    async startSyncingUrlWithAppState() {
+        // our first history entry shall be the one that we end up with when the app loads for the first time
+        window.history.replaceState(null, '', this.createUrlFromState())
+        this.queryStore.register(() => this.updateUrlFromState())
+        this.mapStore.register(() => this.updateUrlFromState())
```

Comment:
Here we listen to updates of the query- and map store. This means whenever some user interaction updates both we would get two new history entries. This would probably not be what we want, but is a limitation of our current approach. 
To work around this we already have the 'ignoreStateUpdates' flag in 'createUrlFromState', because this is already the first case where we want to do multiple state updates for a single 'action'.
An alternative would be pushing history entries for certain user interactions explicitly instead of blindly reacting to these events here.